modules/auth/sso/oauth2.go

```diff
@@ -91,6 +93,35 @@ func (o *OAuth2) userIDFromToken(ctx *macaron.Context) int64 {
 		}
 		return uid
 	}
+
+	if username, token, err := base.BasicAuthDecode(tokenSHA); err == nil && len(token) > 0 && len(username) > 0 {
+		t, err := models.GetAccessTokenBySHA(tokenSHA)
+		if err != nil {
+			if models.IsErrAccessTokenNotExist(err) || models.IsErrAccessTokenEmpty(err) {
+				log.Error("GetAccessTokenBySHA: %v", err)
+			}
+			return 0
```

Comment:
I know the probabilities are remote, but this could still be a valid OAuth2 token it seems. I don't know what to think. Is this still considered an OAuth2 token after basic auth decoding?

Reply:
We're making up new ways to authenticate here. We don't have to support oauth2 for this - so let's not.

Reply:
It wasn't a proposal, I'm just puzzled about the new code. I don't understand it very well. 😅